tests/integration_tests/chat_models/test_anthropic.py

```diff
@@ -0,0 +1,72 @@
+"""Test Anthropic API wrapper."""
+from typing import Generator
+
+import pytest
+
+from langchain.callbacks.base import CallbackManager
+from langchain.chat_models.anthropic import AnthropicChat
+from langchain.schema import AIMessage, BaseMessage, HumanMessage, LLMResult
+from tests.unit_tests.callbacks.fake_callback_handler import FakeCallbackHandler
+
+
+def test_anthropic_call() -> None:
+    """Test valid call to anthropic."""
+    chat = AnthropicChat(model="bare-nano-0")
+    message = HumanMessage(content="Hello")
+    response = chat([message])
+    assert isinstance(response, AIMessage)
+    assert isinstance(response.content, str)
+
+
+def test_anthropic_streaming() -> None:
+    """Test streaming tokens from anthropic."""
+    chat = AnthropicChat(model="bare-nano-0", streaming=True)
+    message = HumanMessage(content="Hello")
+    response = chat([message])
+    assert isinstance(response, AIMessage)
+    assert isinstance(response.content, str)
+
+
+def test_anthropic_streaming_callback() -> None:
+    """Test that streaming correctly invokes on_llm_new_token callback."""
+    callback_handler = FakeCallbackHandler()
+    callback_manager = CallbackManager([callback_handler])
+    chat = AnthropicChat(
+        streaming=True,
+        callback_manager=callback_manager,
+        verbose=True,
+    )
+    message = HumanMessage(content="Write me a sentence with 100 words.")
+    result = chat([message])
+    assert callback_handler.llm_streams > 1
+
+
+@pytest.mark.asyncio
+async def test_anthropic_async_streaming_callback() -> None:
+    """Test that streaming correctly invokes on_llm_new_token callback."""
+    callback_handler = FakeCallbackHandler()
+    callback_manager = CallbackManager([callback_handler])
+    chat = AnthropicChat(
+        streaming=True,
+        callback_manager=callback_manager,
+        verbose=True,
+    )
+    chat_messages: list[BaseMessage] = [HumanMessage(content="How many toes do dogs have?")]
+    result: LLMResult = await chat.agenerate([chat_messages])
+    assert callback_handler.llm_streams > 1
+    assert isinstance(result, LLMResult)
+    for response in result.generations[0]:
+        assert isinstance(response.message, AIMessage)
```

Comment:
The OpenAI tests look like

```
assert isinstance(response, ChatGeneration)
assert isinstance(response.text, str)
assert response.text == response.message.content
```

which should fix the listing complaints.

The point around fixing the chat and LLM result inheritance is a good one and agree it's beyond the scope of this PR

Reply:
Sounds good. Rewritten as such.